include/revng/Support/IRHelpers.h

```diff
@@ -1045,7 +1045,8 @@ inline llvm::Instruction *nextNonMarker(llvm::Instruction *I) {
 inline llvm::CallInst *getFunctionCall(llvm::Instruction *T) {
   revng_assert(T && T->isTerminator());
   llvm::Instruction *Previous = getPrevious(T);
-  while (Previous != nullptr && isMarker(Previous)) {
+  while (Previous != nullptr
+         && (isMarker(Previous) || isCallTo(Previous, "abort"))) {
```

Comment:
The IR is the following one:
```llvm
bb.__lock.0x1b_L0_ft__cloned:                     ; preds = %bb.__lock.0x8_L0_ft__cloned
  %21 = getelementptr i8, i8* %0, i64 -32, !dbg !76574
  %22 = bitcast i8* %21 to i64*, !dbg !76574
  store i64 4208214, i64* %22, align 1, !dbg !76574, !alias.scope !76557, !noalias !76555
  call void @function_call(i8* null, i8* null, %struct.PlainMetaAddress { i32 0, i16 0, i16 4, i64 4208214 }, i64* null, i8* null), !dbg !76575
  call void @abort(), !dbg !76576
  unreachable
}
```
The reason for this was that `isFunctionCall(BB)` would return false, since `abort` is not a marker; and this would prevent us from telling if the callee was noreturn or not. We had discussed this, and eventually we had agreed to make `abort` a marker. In fact, considered that markers (like `function_call`, `newpc`) are not meant to do anything in particular, `abort` does really nothing but signaling that the callee won't return.

Reply:
I see, this is a flaw in the loose conventions in the IR we work on. We should better define it.

However, `nextNonMarker` is not the right place to handle this. You're basically treating `abort` as a marker, while markers are no-op functions that tag code around them. `abort` is definitely not a marker.

Move the check in `isFunctionCall`.

Reply:
It is handled in `getFunctionCall`, not in `nextNonMarker`. `isFunctionCall` simply calls `getFunctionCall`, thus it is the correct place to handle that.